core/txpool/legacypool/list.go

```diff
@@ -148,6 +138,8 @@ func (m *SortedMap) filter(filter func(*types.Transaction) bool) types.Transacti
 		m.cacheMu.Lock()
 		m.cache = nil
 		m.cacheMu.Unlock()
+		// If transactions were removed, the heap and cache are ruined
```

Comment:
I think your change diverges with the original intention.

In the old implementation, the `SortedMap.filter` only removes the transactions and de-validates the cache. The reheap is deliberately separated into another function.

This allows us to do several consecutive filtering and apply a single reheap operation at the end.

Reply:
Get it, great explanation, thanks. and changed.